backpack/extensions/firstorder/batch_grad/batch_grad_base.py

```diff
@@ -15,26 +16,48 @@ class BatchGradBase(FirstOrderModuleExtension):
     In this case, the method is not overwritten by this class.
     """
 
-    def __init__(self, derivatives, params=None):
+    def __init__(self, derivatives, params):
         """Initializes all methods.
 
         If the param method has already been defined, it is left unchanged.
 
         Args:
             derivatives(backpack.core.derivatives.basederivatives.BaseParameterDerivatives):# noqa
                 Derivatives object assigned to self.derivatives.
-            params (list[str]): list of strings with parameter names. Defaults to None.
+            params (list[str]): list of strings with parameter names.
+                For each, a method is assigned.
         """
         self.derivatives = derivatives
         for param_str in params:
             if not hasattr(self, param_str):
-                self.__setattr__(param_str, self._make_function(param_str))
+                self.__setattr__(param_str, self._make_param_function(param_str))
```

Comment:
more a personal choice to avoid dunder methods, but does `setattr(self, param_str, self._make_param_function(param_str))` work, too?